File: app/src/main/java/mega/privacy/android/app/middlelayer/iab/MegaSku.java
package mega.privacy.android.app.middlelayer.iab;


/**
 * Generic SKU object, used to unify corresponding platform dependent purchase object.
 *
 * In HMS, it's ProductInfo.
 * In GMS, it's SkuDetails.
 */
public class MegaSku {

    /**
     * SKU of the product.
     */
    private String sku;

    /**
     * Price of the sku in corresponding platform.
     */
    private long priceAmountMicros;

    /**
     * Currency code, used to format price.
     */
    private String priceCurrencyCode;

    public MegaSku(String sku, long priceAmountMicros, String priceCurrencyCode) {
        this.sku = sku;
        this.priceAmountMicros = priceAmountMicros;
        this.priceCurrencyCode = priceCurrencyCode;
    }

    public String getSku() {
        return sku;
    }

    public long getPriceAmountMicros() {
        return priceAmountMicros;
    }

    public String getPriceCurrencyCode() {
        return priceCurrencyCode;
    }
}

File: domain/src/main/kotlin/mega/privacy/android/domain/entity/account/MegaSku.kt
package mega.privacy.android.domain.entity.account

/**
 * Generic SKU object, used to unify corresponding platform dependent purchase object.
 *
 * In HMS, it's ProductInfo.
 * In GMS, it's SkuDetails.
 */
data class MegaSku(
    /**
     * SKU of the product.
     */
    val sku: String,
    /**
     * Price of the sku in corresponding platform.
     */
    val priceAmountMicros: Long,
    /**
     * Currency code, used to format price.
     */
    val priceCurrencyCode: String
)